resources/aws/instance.go

```diff
@@ -161,6 +161,14 @@ func (i *Instance) Delete() error {
 		return microerror.MaskAny(err)
 	}
 
+	if err := i.Clients.EC2.WaitUntilInstanceTerminated(&ec2.DescribeInstancesInput{
```

Comment:
This can take arbitrarily long (sometimes up to 30 secs), are we sure we need to wait? What's the advantage?

I think it would be better to instead not re-use instances in the *Shutting Down* state.